index.js

```diff
@@ -5,14 +5,31 @@
 process.chdir(__dirname);
 
 // Perform node version check before loading any other files or modules
-// Doing this check as soon as possible allows us to avoid ES6 parser errors or other issues
+// Doing this check as soon as possible allows us to avoid ES6 parser errors or
+// other issues
+// Try to display warnings nicely, but gracefully degrade if anything goes wrong
```

Comment:
I would just add `process.exit(1)` if the version is not satisfied.

Reply:
I thought about it, wondered why we didn't do that already. I guess it comes down to: do we keep the third commit or not?
If so, then yes, app is broken under Node 4, so might as well `exit`.
If not, then it will be compatible until it's not, and we don't have to `exit`.

I am fine with exiting, and we have until v3 to change our mind :)

Reply:
I'd rather exit because I wouldn't want it to crash randomly because we make use of a newer feature.

Reply:
Done:

<img width="543" alt="screen shot 2017-11-22 at 15 56 05" src="https://user-images.githubusercontent.com/113730/33149468-d7c6a84e-cf9d-11e7-80bb-c92478a1ed92.png">

Switched to errors instead of warnings.

Reply:
Well, the third commit shouldn't be part of this PR. It has nothing to do with setting our recommended version of node, it's changing the actual code. I don't have anything against it in general, but I don't think it should be in this PR.

Reply:
I can move this commit to another PR when the Node setting is ready to go (i.e. when we're prepping for v3), but until then I'll keep it in there if you don't mind, as it's easier to test without having to duplicate the other commits, and I don't want to create yet another noisy PR in the list 😅